pkg/ccl/backupccl/backup_destination.go

```diff
@@ -315,7 +318,19 @@ func readLatestFile(
 		return "", err
 	}
 	defer collection.Close()
-	latestFile, err := collection.ReadFile(ctx, latestFileName)
+
+	// First try reading from the latest-history directory. If the backup
+	// is from an older version, it may not exist there yet so try reading
+	// in the base directory if the first attempt fails.
+	latestFile, err := collection.ReadFile(ctx, latestHistoryDirectory)
```

Comment:
following from my comment above `latestHistoryDirectory` it should be this method that is responsible for name.